sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -271,6 +272,18 @@ export class AssessmentController {
           );
           return job.complete();
         }
+        const application = assessment.application;
+        // Only check for previousDateChangedReportedAssessment only if there doesn't exist one and the application is in Completed state.
```

Comment:
Intention of asking you to add a comment is to explain the reason behind it. e.g idempotency.

Reply:
Can you please help me with a suitable comment here.

Reply:
```ts
// Previous date change reported assessments can only exist for completed applications
        // with at least one disbursement sent to federal.
        // Hence applications that are not completed cannot have previous date change reported assessments.
        // To ensure the implementation to be idempotent update previous date change reported
        // only if it is not updated already.
```